pkgs/development/libraries/qt-6/modules/qtbase.nix

```diff
@@ -174,35 +177,44 @@ stdenv.mkDerivation rec {
     EventKit
     GSS
     MetalKit
-  ] ++ lib.optional libGLSupported libGL;
+  ] ++ lib.optionals libGLSupported [
+    libGL
+  ] ++ lib.optionals stdenv.hostPlatform.isMinGW [
+    vulkan-headers
+    vulkan-loader
+  ];
 
-  buildInputs = [
+  buildInputs = lib.optionals (!stdenv.hostPlatform.isMinGW) [
     at-spi2-core
-  ] ++ lib.optionals (!stdenv.isDarwin) [
+  ] ++ lib.optionals (!stdenv.isDarwin && !stdenv.hostPlatform.isMinGW) [
     libinput
   ] ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [
     AppKit
     CoreBluetooth
   ]
   ++ lib.optional withGtk3 gtk3
   ++ lib.optional developerBuild gdb
-  ++ lib.optional (cups != null) cups
-  ++ lib.optional (libmysqlclient != null) libmysqlclient
-  ++ lib.optional (postgresql != null) postgresql;
+  ++ lib.optional (cups != null && !stdenv.hostPlatform.isMinGW) cups
+  ++ lib.optional (libmysqlclient != null && !stdenv.hostPlatform.isMinGW) libmysqlclient
+  ++ lib.optional (postgresql != null && !stdenv.hostPlatform.isMinGW) postgresql;
 
   nativeBuildInputs = [ bison flex gperf lndir perl pkg-config which cmake xmlstarlet ninja ]
     ++ lib.optionals stdenv.isDarwin [ moveBuildTree ];
 
   propagatedNativeBuildInputs = [ lndir ];
 
+  depsBuildBuild = lib.optionals stdenv.hostPlatform.isMinGW [
+    pkgsBuildBuild.qt6.qtbase
+  ];
```

Comment:
Doesn't this cause inf recursion when doing native compilation on mingw?

Reply:
I have no idea. I didn't know that you can use mingw as a build platform. I can modfiy the logic to `(stdenv.hostPlatfrom.isMinGW && (stdenv.hostPlatfrom != stdenv stdenv.buildPlatform))` if there is concern.

Reply:
>  I didn't know that you can use mingw as a build platform.

I don't know either, but we want to support that right?

>  I can modfiy the logic to `(stdenv.hostPlatfrom.isMinGW && (stdenv.hostPlatfrom != stdenv stdenv.buildPlatform))` if there is concern.

More like `(stdenv.hostPlatfrom.isMinGW && (!stdenv.buildPlatform.canExecute stdenv.hostPlatform))`. But why is mingw special? Shouldn't the same apply to cross compilation to all platforms?

Reply:
> But why is mingw special? Shouldn't the same apply to cross compilation to all platforms?
https://doc.qt.io/qt-6/cmake-variable-qt-host-path.html
You are correct. Will update the logic.

Reply:
Fixed